flume-ng-node/src/main/java/org/apache/flume/node/Application.java

```diff
@@ -273,6 +279,12 @@ public static void main(String[] args) {
       if (commandLine.hasOption('z') || commandLine.hasOption("zkConnString")) {
         isZkConfigured = true;
       }
+
+      // initialize class loader factory from the plugins path
+      // this will isolate Flume agent plugins to their own classloader
+      String pluginsPath = commandLine.getOptionValue("plugins-path");
+      PluginsClassLoaderFactory.initialize(pluginsPath);
```

Comment:
plugins-path arg is set only when --use-plugins-classloader is set to true. 
In normal flow, the plugins-path will be null resulting into all plugins being loaded by same Application class loader.